### PR TITLE
Using CMake to build libraqm

### PR DESCRIPTION
**Pull request by alfarsi92**
In this phase:
Building successfully in Linux (Ubuntu) and Windows (MSYS2, Cygwin, Visual studio 2013 softwares)

Next Phase:
Testing & solve the warning messages in gtk-doc in Linux

### CODE REVIEW

File: CMakeLists.txt~ [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# minimum required vertion of cmake
```
Comment (khaledhosny) [inline CMakeLists.txt~]: This file should not be committed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,88 @@
+<<<<<<< HEAD
```
Comment (khaledhosny) [inline CMakeLists.txt]: Bad merge conflict resolving.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+cmake_minimum_required (VERSION 3.2)
+
+# project name
+project (raqm)
+
+##############################3
+
+#enable compiler warning
+if(CMAKE_COMPILER_IS_GNUCC)
+add_definitions (-Wall  -Wno-portability)
```
Comment (khaledhosny) [inline CMakeLists.txt]: Autotools build enables more compiler warnings, we should do the same here.

File: docs/CMakeLists.txt~ [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# project name
```
Comment (khaledhosny) [inline docs/CMakeLists.txt~]: And this file as well.

File: documentation for msys.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Msys2:
```
Comment (khaledhosny) [inline documentation for msys.txt]: What is this file?

**Comment by khaledhosny** [comment]
I don’t know why you are including all these third party files, but this is a no no. If this is really needed, then this should live in an external repository, say raqm-windows, that provides an easy to use way to build Raqm on Windows.

**Comment by khaledhosny** [comment]
The Travis build is failing, this needs to be fixed as well.

**Comment by Fahad-Alsaidi** [comment]
> I don’t know why you are including all these third party files, but this is a no no. If this is really needed, then this should live in an external repository, say raqm-windows, that provides an easy to use way to build Raqm on Windows.

I am afraid that external repository get rusted by time. What about offering zip file the same way scribus offers for windows?

**Comment by Fahad-Alsaidi** [comment]
@alfarsi92 do you want to continue the work and  modify the pull request As Khaled requested?

**Comment by khaledhosny** [comment]
The same would happen here if no one is maintaining the code, and in its current form it is unmaintainable.

File: documentation for msys.txt (hunk 5 above)
Comment (Fahad-Alsaidi) [inline documentation for msys.txt]: This is instructions how to built raqm with Msys2. I think it should move to readme.txt

**Pull request closed without merging**